### ISSUE TITLE
Blazor .NET 8 tracking issue

### ISSUE THREAD

**Issue by guardrex** [dotnet/AspNetCore.Docs#28161]
## 8.0 Post GA

***NOTE***: Probably best to have the PWA doc reviewed by the PU for any other 8.0 PWA changes.

Status | Issue | PR | Item
:---: | :---: | :---: | ---
🚧<br>**IN PROGRESS**<br>👷 | #32020 | &mdash; | Update Blazor Web App glob/loc approach
🟢 | #31691 | &mdash; | Update section on passing tokens in Blazor Web Apps<br><br>***Waiting on guidance from Halter/Jeremy.***
🟢 | #31381 | &mdash; | Additional or different bearer token remarks for WASM security
✔️ | #32008 | #32028 | Cookie security for Call web API article
✔️ | #32082 | #32083 | Describe dependency on DBContext (vs DBContextFactory) for Blazor Identity UI
✔️ | #32077 | #32081 | Cover nonce update for token refresh
✔️ | #31402 | #32048 | Interactive rendering required for `NotAuthorized` component/content to work ... and I'll probably work this in tandem with `NotFound` content behavioral differences between ASP.NET Core pipeline on the server and WASM on the client.
✔️ | #32054 | #32060 | Blazor framework static files served via endpoint routing
✔️ | #31383 | #31431 | Account confirmation+PW recovery article for BWA/Identity components **UPDATE**: I asked multiple times for PU review, and it was never followed up on by PU engineers.
✔️ | #31931 | #32035 | Using fallback authorization policy breaks Blazor Server after migrating to .NET 8 
❌ | #31899 | &mdash; | Cookie example does not work for .Net 8 template because the state container does not instantiate in user scope ***Dup tracked by #39691.***
✔️ | #30002 | #31894 | Dynamic culture approach updates (last one is maintaining state across interactivity location change)
✔️ | #31973 | #31991 | Security aspects in the Call web API article
✔️ | #31887 | #31961 | Cover web API approaches for the BWA Auto render mode scenario
✔️ | #31389 |  #31888 | Samesite + addl config and guidance for WASM+Identity different domain hosting
✔️ | #31379 | #31934 | QuickGrid article API doc cross-links
✔️ | #31205 | https://github.com/dotnet/blazor-samples/pull/136 | BlazorWebAssemblyStandaloneWithIdentity uses cookie auth but no antiforgerytoken
✔️ | #31741 | #31873 | Adopt 8.0 conventions for inline Blazor forms
✔️ | #31740 | #31867 | Mention that App.razor always renders statically
✔️ | #31811 | #31852 | Consider documenting additional WebAssembly (runtime) startup callbacks
✔️ | #31672 | #31850 | NSwag for WASM+Identity article+sample
✔️ | #31840 | #31841 | BWA+OIDC signout from secure page death 💀 loop
✔️ | #31265 | #31555 | Companion article for new Blazor OIDC sample with Aspire
✔️ | #31045 | https://github.com/dotnet/blazor-samples/pull/140<br>#31839 | Role claims guidance in standalone WASM w/Identity article
✔️ | #31688 | #31693 | Blazor 8.0 offline discussion updates
✔️ | #31796 | #31823 | Cross-link SignalR request credentials guidance
✔️ | #31689 | #31705 #31738 | Transient disposable section updates
✔️ | #30443 | #31638 #31639 #31671 #31677 | Follow-up debug items
✔️ | #31620 | #31622 | Blazor content follow-up updates for 8.0
✔️ | #31613 | #31619 | Add troubleshooting guidance to WASM+Identity article
✔️ | #31586 | #31617 | Update WASM+Entra groups and roles article
✔️ | #31554 | #31569 | Drop `[Parameter]` for query string-supplied params
✔️ | #31562 | #31566 | Terminology/guidance updates on server versus `.Client` projects of a BWA
✔️ | #31561 | #31563 | Update PWA guidance
✔️ | #31543 | #31544 | Add clarification on interactive root component remark
✔️ | #31429 | #31445 | Integration article updates
✔️ | #31496 | #31503 | Improve auth state provider guidance for BWAs
✔️ | #31316 | #31488 | BlazorWebApp .Net 8 App State management
✔️ | #31384 | #31500 | Server-side authentication mentions SignalIR - is this still always true in 8.0?
✔️ | #31228 | #31477 | Clarify Auto render mode behavior
✔️ | #31428 | #31438 #31476 | Describe InputSelect form behavior for SSR
✔️ | #31023 | #31427 | Blazor CRUD/Quickgrid scaffolder (preview)
✔️ | #31393 | #31425 | RazorComponentResult<T> doesn't render either the Blazor or MVC Layout
✔️ | #31403 |  #31404 #31405 | Apply a render mode programmatically section
✔️ | #30043 | #31390 | App base path updates |
✔️ | #31267 | #31382 | Further work on client-side services that fail on prerendering
✔️ | #31351 | #31353 | Prerendering and interactive/enhanced routing
✔️ | #31271 | #31282 (Scaffolding)<br>#31378 (Blazor Security) | Cover Razor component Identity UI for BWAs
✔️ | #31194 | #31214 | Update token auth guidance for WASM+Identity article
✔️ | #31048 | #31145 | Document how to display a loading UI for interactive WebAssembly rendering
✔️ | #31029 | https://github.com/dotnet/blazor-samples/pull/133<br>#31236<br>#31268<br>#31274<br>#31283<br>#31286<br>https://github.com/dotnet/blazor-samples/pull/154<br>#31341<br>https://github.com/dotnet/blazor-samples/pull/157<br>#31364 | BWA/WASM snippet sample app and article cross-links
✔️  | #31191 | #31192 | Fix up Razor component directive ordering
✔️ | #31089 | #31183 | Terminology consistency updates
✔️ | #31081 | #31131 | Improve AdditionalAssemblies coverage
✔️ | #31170 | #31173 | Layout namespace for Blazor Web Apps<br><br>Asking the PU if they want the BWA template to make layouts more readily consumable with a namespace in the `_Imports` file: https://github.com/dotnet/aspnetcore/issues/52478
✔️ | #31153 | #31155 | Cross-link API docs for new API.
✔️ | #31117 | #31118 | Update enhanced navigation remarks
✔️ | #31107 | #31109 | Additional Blazor start updates for 8.0
✔️ | #31105 | #31106 | Update CSP guidance **UPDATE**: The CSP guidance is fine. The PR adds some MIA coverage unrelated to 8.0 that we need.
✔️ | #31078 | #31079 | Suggestion: Enhance section on changing the default render mode for an app
✔️ | #31058 | #31073 | New article from Javier's content on HTTP Caching Issues When Upgrading Blazor Applications Across Major Versions
✔️ | #31066 | #31090 | New article from Steve's content on Static server-side rendering for Blazor component libraries article
✔️ | #31056 | #31065 | Cascading auth state services registration with `AddCascadingAuthenticationState` to replace the `CascadingAuthenticationState` component in >=8.0
✔️ | #31043 | #31063 | Security considerations for BWA SSR
✔️ | #31039 | #31049 | Update from DLL to WASM files 8.0
✔️ | &mdash; | #31024 | Update `ms.date` metadata.
✔️ | &mdash; | #31004 | Activate the Prereqs sections in the Blazor-SignalR and ToDo List tutorials.
✔️ | &mdash; | https://github.com/dotnet/blazor-samples/pull/123 | Update package refs at `dotnet/blazor-samples` to 8.0 release packages.
✔️ | &mdash; | #31004 | Final updates to *What's New* and *Migration* content.
✔️ | #31025 | #31057 | Environments and client services with prerendering
✔️ | &mdash; | #31024 | Update the old content alert version and add the INCLUDE to the new QuickGrid article and in the JavaScript JSImport/JSExport interop article (= 7.0).
❌ | #30820 | &mdash; | Streaming rendering with Azure hosting ***Referred to the PU as a support request.*** I'll re-open and work the issue if they say a remark should be added on hosted (non-Blazor) APIs that might need to flush their response.
❌ | #30994 | &mdash; | Blazor BWA external/social login guidance: This will be based on a Jeremy sample app that will be created within the next week or two. *Closed as not planned. We won't have a sample with a specific social auth provider implemented.* https://github.com/dotnet/AspNetCore.Docs/issues/30994#issuecomment-1861785854
❌  | #30392 | &mdash; | **MOVED to .NET 9:** Multiple hosted BWAs in an ASP.NET Core app | I'm still unclear tho if we'll have multiple hosted `.Client` projects in a BWA, if such a scenario is even sane.

## Derived from 8.0 issues/PRs not on the roadmap

🟢 = Ready&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; ✔️ = Done&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; 🛑 = Wait on PU to finish&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; ❌ = No action required

Status | Live | Issue | PR | Item | PU Links/<br>Blog
:---: | :---: | :---: | :---: | --- | :---:
✔️ | [Live (first two sections)](https://learn.microsoft.com/en-us/aspnet/core/blazor/security/webassembly/standalone-with-identity) | #29363 #30394 | #31004 | Identity API endpoints (`MapIdentityApi<TUser>()`), Identity Razor components, token auth, etc. | <ul><li>[Likness feature remarks in GH issue](https://github.com/dotnet/aspnetcore/issues/42158#issuecomment-1481742187)</li><li>[Likness blog post](https://devblogs.microsoft.com/dotnet/improvements-auth-identity-aspnetcore-8/)</li><li>[Razor components for auth](https://github.com/dotnet/aspnetcore/issues/48786)</li><li>[List of PU repo Identity issues](https://github.com/dotnet/aspnetcore/issues?q=label%3Afeature-token-identity+sort%3Aupdated-desc)</li><li>[Identity API endpoints starting with `/register` and `/login` (blog post)](https://devblogs.microsoft.com/dotnet/asp-net-core-updates-in-dotnet-8-preview-4/#identity-api-endpoints)</li><li>[APIs for identity management](https://github.com/dotnet/aspnetcore/issues/47288)</li></ui> **UPDATE**: The Blazor issue will be worked after #29363 and #30066 are merged. There's some token coverage now in [Jeremy's blog post](https://devblogs.microsoft.com/dotnet/whats-new-with-identity-in-dotnet-8/).
❓❓❓<br>**Requires research in PU issues/PRs**<br>❓❓❓ | &mdash; | &mdash; | &mdash; | Auto render mode behavior in mixed render mode situations | https://github.com/dotnet/aspnetcore/issues/50686 and https://github.com/dotnet/aspnetcore/issues/50677
✔️ | [Live](https://learn.microsoft.com/en-us/aspnet/core/blazor/security/server/?view=aspnetcore-8.0&tabs=visual-studio) | #30976 | #30977 | Provide a set of Blazor authentication components in the Blazor project template | <ul><li>https://github.com/dotnet/aspnetcore/issues/48786</li><li>https://github.com/dotnet/aspnetcore/pull/50722</li><li>https://github.com/dotnet/aspnetcore/pull/51134</li><li>[Project Template bits](https://github.com/dotnet/aspnetcore/tree/main/src/ProjectTemplates/Web.ProjectTemplates/content/BlazorWeb-CSharp/BlazorWeb-CSharp/Components/Account)</li></ul>
✔️ | [Live](https://learn.microsoft.com/en-us/aspnet/core/blazor/security/webassembly/standalone-with-identity?view=aspnetcore-8.0) | #30944 | #30957 | Standalone Blazor WASM with Identity
✔️ | [Live](https://learn.microsoft.com/en-us/aspnet/core/blazor/javascript-interoperability/static-server-rendering?view=aspnetcore-8.0) | #30921 | #30922 | Using JavaScript with Blazor Static Server Rendering
✔️ | [Live](https://learn.microsoft.com/en-us/aspnet/core/blazor/fundamentals/startup?view=aspnetcore-8.0) | #30812 | #30819 | Additional Blazor start configuration
✔️ | [Live](https://learn.microsoft.com/en-us/aspnet/core/blazor/fundamentals/startup?view=aspnetcore-8.0#javascript-initializers) | #30918 | #30920 | JavaScript initializers for BWA | [PU Issue](https://github.com/dotnet/aspnetcore/issues/50187) [PU PR](https://github.com/dotnet/aspnetcore/pull/50860)
✔️ | [Live](https://learn.microsoft.com/en-us/aspnet/core/blazor/forms/validation?view=aspnetcore-8.0#server-validation-with-a-validator-component) | #30055 | #30925 | Server validation with validator components for BWA |
✔️ | [Live](https://learn.microsoft.com/en-us/aspnet/core/blazor/host-and-deploy/webassembly?view=aspnetcore-8.0#trim-net-il-after-ahead-of-time-aot-compilation) | #30535 | #30339 #30886 | Add guidance trimming IL after AOT | I'm 👂 on [an issue in the runtime repo](https://github.com/dotnet/runtime/issues/93017) for what else, if anything, they want to say about it.
✔️ | &mdash; | #30338 | #30374 | Update form custom validation examples to use `EditContext.IsValid` API. | 
✔️ | [Live](https://learn.microsoft.com/en-us/aspnet/core/blazor/host-and-deploy/webassembly-deployment-layout?view=aspnetcore-8.0) | #30416 | #30419 | Drop multipart bundling package support | Support for the package is dropped per https://github.com/dotnet/aspnetcore/issues/36978#issuecomment-1439283893. The article will remain for 8.0+ with the updates made on the PR.
✔️ | [Live](https://learn.microsoft.com/en-us/aspnet/core/blazor/debug?view=aspnetcore-8.0&tabs=visual-studio) | #30170 | #30440 | Blazor Debug article updates | This received a full overhaul and update to include generalizing the topic with server-side debugging  guidance ... BWA (>=8.0) and Blazor Server apps (<8.0). The .NET 8 updates are mostly done. I will open a follow-up issue for a few items, including some trouble that I had with MS Edge browser debugging (can't hit breakpoints).
✔️ | [Live 1](https://learn.microsoft.com/en-us/aspnet/core/blazor/components/prerender?view=aspnetcore-8.0)<br>[Live 2](https://learn.microsoft.com/en-us/aspnet/core/blazor/components/integration?view=aspnetcore-8.0) | #30341 | #30455 #30514 #30791 #30799 | Split and update Blazor Prerendering & Integration article |
✔️ | [Live](https://learn.microsoft.com/en-us/aspnet/core/blazor/host-and-deploy/webassembly?view=aspnetcore-8.0#ahead-of-time-aot-compilation) | #30325 | #30326 | Add "Jiterpreter" to description |
✔️ | &mdash; | #30251 | #30253 | Version hosted WASM <8.0 |
✔️ | [Live](https://learn.microsoft.com/en-us/aspnet/core/blazor/host-and-deploy/webassembly?view=aspnetcore-8.0#compression) | #30213 | #30214 | Update Blazor WASM compression flag | [PU Issue](https://github.com/dotnet/aspnetcore/issues/50451) [PU PR](https://github.com/dotnet/sdk/pull/31559)
✔️ | [Live](https://learn.microsoft.com/en-us/aspnet/core/blazor/fundamentals/handle-errors?view=aspnetcore-8.0#error-boundaries) | #30127 | #30128 | Error boundary for 8.0/BWA | https://github.com/dotnet/aspnetcore/issues/49679
✔️ | [Live](https://learn.microsoft.com/en-us/aspnet/core/blazor/fundamentals/routing?view=aspnetcore-8.0#enhanced-navigation-and-form-handling) | #30322 #30876 | #30336 #30884 | "Enhanced" navigation & API for enhanced page refresh | https://github.com/dotnet/aspnetcore/issues/49414 https://github.com/dotnet/aspnetcore/issues/50014
✔️ | [Live](https://learn.microsoft.com/en-us/aspnet/core/blazor/fundamentals/startup?view=aspnetcore-8.0#configure-the-net-webassembly-runtime) | #30320 | #30321 | Have a way to configure dotnet runtime for blazor apps | https://github.com/dotnet/aspnetcore/issues/49264 https://github.com/dotnet/aspnetcore/pull/49420
✔️ | &mdash; | #30323 | #30324 | Remove suppress-error from script | https://github.com/dotnet/aspnetcore/issues/49053
✔️ | &mdash; | #30328 #30711 | #30329 | Router's NotFound content is never used in new Web project style | https://github.com/dotnet/aspnetcore/issues/48983 https://github.com/dotnet/aspnetcore/pull/49801
❌ | &mdash; | &mdash; | &mdash; | Default WebAssembly loading progress indicator doesn't apply in Blazor Web template | **No action needed.** This was always indicated as a for-WASM-only scenario in the *Startup* doc. [PU Issue](https://github.com/dotnet/aspnetcore/issues/49056) [PU PR](https://github.com/dotnet/aspnetcore/pull/49801)
✔️ | [Live](https://learn.microsoft.com/en-us/aspnet/core/blazor/fundamentals/signalr?view=aspnetcore-8.0#ihttpcontextaccessorhttpcontext-in-razor-components) | &mdash; | #30576 | Blazor SSR APIs for accessing HttpContext without IHttpContextAccessor | [PU Issue](https://github.com/dotnet/aspnetcore/issues/48769) [PU PR](https://github.com/dotnet/aspnetcore/pull/50253) ***AND*** #30575
✔️ | &mdash; | &mdash; | #30576 | Enhanced form handling will be opt-in | https://github.com/dotnet/aspnetcore/pull/50551 https://github.com/dotnet/AspNetCore.Docs/pull/30279#discussion_r1321920797
✔️ | &mdash; | &mdash; | #30576 | Blazor WASM empty template guidance| 
✔️ | [Live](https://learn.microsoft.com/en-us/aspnet/core/blazor/fundamentals/routing?view=aspnetcore-8.0#enhanced-navigation-and-form-handling) | #30372 | #30576 | Blazor APIs to preserve content between enhanced navigations | https://github.com/dotnet/aspnetcore/pull/50437
✔️ | [Live](https://learn.microsoft.com/en-us/aspnet/core/blazor/components/cascading-values-and-parameters?view=aspnetcore-8.0) | #30093 #30947 | #30095 #30958 | Root-level cascading values | [Blog](https://devblogs.microsoft.com/dotnet/asp-net-core-updates-in-dotnet-8-preview-7/#register-root-level-cascading-values), [PU Issue](https://github.com/dotnet/aspnetcore/issues/49104), [PU PR](https://github.com/dotnet/aspnetcore/pull/49204) **NOTE:** Didn't include a fully-working service example that changes the cascading param value with notification or the factory-based approach. Will open an issue later to add these.
✔️ | [Live](https://learn.microsoft.com/en-us/aspnet/core/blazor/components/virtualization?view=aspnetcore-8.0#empty-content) | #30094 | #30096 | Virtualize empty content | [Blog](https://devblogs.microsoft.com/dotnet/asp-net-core-updates-in-dotnet-8-preview-7/#new-emptycontent-parameter-for-virtualize), [PU Issue](https://github.com/dotnet/aspnetcore/issues/28770), [PU PR](https://github.com/dotnet/aspnetcore/pull/49185)
✔️ | [Live](https://learn.microsoft.com/en-us/aspnet/core/blazor/components/control-head-content?view=aspnetcore-8.0#headoutlet-component) |  #29792 | #29793 | `HeadOutlet` component | &mdash;
❌ | &mdash; | &mdash; | &mdash; | Blazor sample apps might get Open Iconic treatment depending on PU action | [Docs Issue](https://github.com/dotnet/AspNetCore.Docs/issues/29330) ***UPDATE***: Open Iconic is being left in the 7.0 project templates ([WASM](https://github.com/dotnet/aspnetcore/tree/release/7.0/src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/Client/wwwroot/css/open-iconic), [Server](https://github.com/dotnet/aspnetcore/tree/release/7.0/src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/wwwroot/css/open-iconic)), so we'll only be dropping it for doc 8.0+ sample apps.
✔️ | [Live: Blazor SignalR Fundamentals](https://learn.microsoft.com/en-us/aspnet/core/blazor/fundamentals/signalr?view=aspnetcore-8.0#configure-signalr-timeouts-and-keep-alive-on-the-client)<br>[Live: Host & Deploy WASM](https://learn.microsoft.com/en-us/aspnet/core/blazor/host-and-deploy/webassembly?view=aspnetcore-8.0#global-deployment-and-connection-failures)<br>[Live: Host & Deploy Server](https://learn.microsoft.com/en-us/aspnet/core/blazor/host-and-deploy/server?view=aspnetcore-8.0#global-deployment-and-connection-failures) | #29296 | #29297 | Setting `ServerTimeout` and `KeepAliveInterval` | [PU Issue](https://github.com/dotnet/aspnetcore/issues/44742) [PU PR](https://github.com/dotnet/aspnetcore/pull/46065)
✔️ | [Live](https://learn.microsoft.com/en-us/aspnet/core/blazor/globalization-localization?view=aspnetcore-8.0&pivots=webassembly#net-globalization-and-international-components-for-unicode-icu-support) | #29317 | #29318 | Custom ICU data loading for WASM | [dotnet/runtime #54486](https://github.com/dotnet/runtime/issues/54486)<br>[dotnet/runtime #80421](https://github.com/dotnet/runtime/pull/80421)<br>[dotnet/AspNetCore.Docs #28368](https://github.com/dotnet/AspNetCore.Docs/pull/28368)
✔️ | [Live](https://learn.microsoft.com/en-us/aspnet/core/blazor/debug?view=aspnetcore-8.0&tabs=visual-studio) | #28940 | #28953 | Hot reload support for instance fields, properties, and events for .NET on WebAssembly ***AND*** Support for symbol servers when debugging .NET on WebAssembly ***AND*** Blazor WebAssembly debugging in Firefox | [FF PU remarks](https://github.com/dotnet/runtime/issues/82290#issuecomment-1434584267)
✔️ | [Live](https://learn.microsoft.com/en-us/aspnet/core/blazor/debug?view=aspnetcore-8.0&tabs=visual-studio) | &mdash; | &mdash; | Update the "preview" guidance for debugging with Firefox and see #28953 | &mdash;
✔️ | [Live](https://learn.microsoft.com/en-us/aspnet/core/blazor/tooling?view=aspnetcore-8.0&pivots=windows#net-webassembly-build-tools) | #28996 #30151 #30923 | #28997 #30152 #30924 | Blazor WASM SIMD support | [Blog](https://devblogs.microsoft.com/dotnet/asp-net-core-updates-in-dotnet-8-preview-3/#simd-enabled-by-default-for-blazor-webassembly-apps) [PU PR](https://github.com/dotnet/AspNetCore.Docs/issues/28996)
✔️ | [Live](https://learn.microsoft.com/en-us/aspnet/core/blazor/javascript-interoperability/import-export-interop?view=aspnetcore-8.0) | #29101 | #29102 | Blazor JS import/export interop updates | 
✔️ | [Live](https://learn.microsoft.com/en-us/aspnet/core/blazor/fundamentals/routing?view=aspnetcore-8.0#query-strings) | #29547 | #29705 | Update [SupplyParameterFromQuery] guidance and examples | [PU PR](https://github.com/dotnet/aspnetcore/pull/48554)
✔️ | [Live](https://learn.microsoft.com/en-us/aspnet/core/blazor/components/lifecycle?view=aspnetcore-8.0#after-component-render-onafterrenderasync) | #30097 | #30098 | Clarify API docs for OnAfterRender | [PU PR](https://github.com/dotnet/aspnetcore/pull/49519)
✔️ | [Live](https://learn.microsoft.com/en-us/aspnet/core/blazor/components/render-modes?view=aspnetcore-8.0#closure-of-circuits-when-there-are-no-remaining-interactive-server-components) | &mdash; | #30576 | Closure of circuits when there are no remaining interactive server components | 
✔️ | [Live](https://learn.microsoft.com/en-us/aspnet/core/blazor/fundamentals/dependency-injection?view=aspnetcore-8.0#inject-keyed-services-into-components) | &mdash; | #30576 | Inject keyed services into components | 
✔️ | [Live](https://learn.microsoft.com/en-us/aspnet/core/blazor/components/event-handling?view=aspnetcore-8.0#support-for-dialog-cancel-and-close-events) | &mdash; | #30576 | Support for dialog cancel and close events | 

## Major updates for full-stack UI *by article*

*Status* doesn't include ...

* Final updates for standalone issues opened for additional specific work.
* Questions for the PU, API doc cross-links, and 8.0 snippet sample cross-links. Questions are tracked by in-text HTML comments that will be checked at (and after) RTM or via the tables/items further down this issue. 

Feature/Task | Status | Docs Issue | Docs PR | Notes
--- | :---: | :---: | :---: | ---
Blazor overview |✔️| #29657 #30285 | #29659 #30353 |
Supported Platforms |✔️| &mdash; | &mdash; | 
Tooling | ✔️ | #29567 | #29568 #29796 #30296 #30358 #30365 #31004 | 
Hosting models/Render modes | ✔️ | #29861 | #30279 #30397 #30576 #31004 | 
Todo tutorial | ✔️ | #29668 | #29669 #29681 #29671 #30296 #30576 #31004 | 
SignalR tutorial (Preview article) | ✔️ | #29679 | #29681 #30296 #30375 #30576 #31004 | 
Project structure | ✔️ | #29565 | #29566 #30359 #30576 #31004 | 
Fundamentals node | ✔️ | #29917 | #29918 #29920 #29928 #29996 #30576 #31004 | 
Components node | ✔️ | #29917 | #29931 #29972 #29996 #30576 #31004 | 
Glob and Loc |✔️| #30002 #30003 | #30004 #30576 | Two tasks done and one remains on #30002. It still needs an approach to integrate server- and client-side dynamic culture approaches for BWA (i.e., the `Auto` render mode scenario).
Forms and Antiforgery | ✔️ | #29728 | #30035 #30078 #30081 #30576 #30834 #30997 #31004 |
File Uploads |✔️| #30111 #30914 | #30117 #30915 | 
File Downloads |✔️| #30130 | #30138 | 
JS interop node |✔️| #30140 | #30171 | 
Call a web API |✔️| #30172 | #30180 | 
Images |✔️| #30181 | #30196 | 
Security |✔️| #30203 | #30210 #30576 | 
State Management |✔️| #30211 | #30227 | 
Debug WASM |✔️| &mdash; | &mdash; | WASM only: No changes needed.
Lazy-load Assemblies WASM |✔️| &mdash; | &mdash; | WASM only: No changed needed.
WASM Native Deps |✔️| &mdash; | &mdash; | WASM only: No changed needed.
Performance |✔️| #30228 | #30262 | 
Test Components |✔️| &mdash; | &mdash; | No changes needed.
PWAs |✔️| #30261 | &mdash; | One line changed WRT hosted WASM on an earlier issue. Otherwise, the PU will need to analyze this content.
Host and Deploy |✔️| #30265 | #30393 |
Blazor Server with EF Core |✔️| #30398 | #30405 | 
Advanced Scenarios |✔️| #30263 | #30264 | 

Other

* Fundamentals > Logging: Although the purpose of remarking on "client-side" for several of the features was merely meant to mean from the original text/purpose that the feature ***also*** works client-side, the section heading use of "client-side" implies that it might not work server-side, when it actually does. Therefore, I'll probably be dropping "client-side" from headings where the feature/example works for both.
* Fundamentals > Routing: https://github.com/dotnet/aspnetcore/issues/48983 pertains to the removal of the `<NotFound>` fragment. I need to run thru the docs and confirm that it's out for 8.0+.
* Components > Class libraries: Check for additional updates per ...
  * https://github.com/dotnet/aspnetcore/issues/47624
  * https://github.com/dotnet/aspnetcore/issues/49880

## Derived from the 8.0 roadmap

🟢 = Ready&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; ✔️ = Done&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; 🛑 = Wait on PU to finish&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; ❌ = No action required

Status | Live | Issue | PR | Item | PU Links/<br>Blog
:---: | :---: | :---: | :---: | --- | :---:
✔️ | [Live](https://learn.microsoft.com/en-us/aspnet/core/blazor/components/quickgrid?view=aspnetcore-8.0) | #28576 | #28622 #28767 #28827 #29817 | QuickGrid productization | [PU Issue](https://github.com/dotnet/aspnetcore/issues/46317) [PU PR](https://github.com/dotnet/aspnetcore/pull/46573)
✔️ | See above | See above | See above | Full stack web UI with Blazor | See above
✔️ | [Live](https://learn.microsoft.com/en-us/aspnet/core/blazor/fundamentals/handle-errors?view=aspnetcore-8.0#handle-caught-exceptions-outside-of-a-razor-components-lifecycle) | #28933 | #28936 | API for dispatching to renderer sync context and capturing exceptions ("Dispatch exceptions to Blazor's `SynchronizationContext`") | [PU Issue](https://github.com/dotnet/aspnetcore/issues/44920) [API Proposal](https://github.com/dotnet/aspnetcore/issues/46068) [PU PR](https://github.com/dotnet/aspnetcore/pull/46074)
✔️ | [Live](https://learn.microsoft.com/en-us/aspnet/core/blazor/components/render-components-outside-of-aspnetcore?view=aspnetcore-8.0) | #28980 | #28981 | Support rendering components outside of the HttpContext | [PU Issue ](https://github.com/dotnet/aspnetcore/issues/38114) [PU PR](https://github.com/dotnet/aspnetcore/pull/46935) [API Design](https://github.com/dotnet/aspnetcore/issues/47018)
✔️ | [Live](https://learn.microsoft.com/en-us/aspnet/core/blazor/fundamentals/routing?view=aspnetcore-8.0#hashed-routing-to-named-elements) | #29105 | #29106 | Hashed routing to named elements | [PU Issue](https://github.com/dotnet/aspnetcore/issues/8393) [PU PR](https://github.com/dotnet/aspnetcore/pull/47320)
✔️ | [Live](https://learn.microsoft.com/en-us/aspnet/core/blazor/security/content-security-policy?view=aspnetcore-8.0) | #29293 | #29294 | Remove unsafe-eval CSP Requirement for Blazor WASM | [PU Issue](https://github.com/dotnet/aspnetcore/issues/37787)
✔️ | [Live](https://learn.microsoft.com/en-us/aspnet/core/blazor/components/sections?view=aspnetcore-8.0) | #28617 | #28660 #29795 | Sections support in Blazor | [PU Issue](https://github.com/dotnet/aspnetcore/issues/28182) [PU Issue](https://github.com/dotnet/aspnetcore/issues/46937) [PU PR](https://github.com/dotnet/aspnetcore/pull/46727)
✔️ | [Live](https://learn.microsoft.com/en-us/aspnet/core/blazor/fundamentals/signalr?view=aspnetcore-8.0#monitor-circuit-activity-blazor-server) | #28982 | #28983 | Ability to monitor circuit activity | [PU Issue](https://github.com/dotnet/aspnetcore/issues/30287)
✔️ | [Live1](https://learn.microsoft.com/en-us/aspnet/core/blazor/hosting-models?view=aspnetcore-8.0#blazor-webassembly) [Live2](https://learn.microsoft.com/en-us/aspnet/core/blazor/host-and-deploy/webassembly?view=aspnetcore-8.0#ahead-of-time-aot-compilation) | #28969 | #28970 | [wasm] Jiterpreter tracking issue runtime | [PU Issue](https://github.com/dotnet/runtime/issues/78428)
✔️ | [Live](https://learn.microsoft.com/en-us/aspnet/core/blazor/host-and-deploy/webassembly?view=aspnetcore-8.0#webcil-packaging-format-for-net-assemblies) | #28964 | #29542 | Publish .NET assemblies in Webcil files runtime / Webcil format replaces multipart bundling | [PU PR](https://github.com/dotnet/runtime/pull/79416)
❓❓❓<br>**Requires research in PU issues/PRs**<br>❓❓❓ | [Live](https://learn.microsoft.com/en-us/aspnet/core/test/hot-reload?view=aspnetcore-8.0#blazor-webassembly) | &mdash; | &mdash; | WASM Hot Reload updates | [Continued improvements to Mono Method Body Replacement runtime (dotnet/runtime #57365)](https://github.com/dotnet/runtime/issues/57365) and [Supported Edits in Edit & Continue (EnC) and Hot Reload on .NET 8 (Roslyn docs)](https://github.com/dotnet/roslyn/blob/main/docs/wiki/EnC-Supported-Edits.md)

## Moved to .NET 9

* [ Better reconnection logic for Blazor Server](https://github.com/dotnet/aspnetcore/issues/32113)
* [Real multithreading](https://github.com/dotnet/aspnetcore/issues/17730)
* [[wasm] Tracking issue for build/bundler changes runtime](https://github.com/dotnet/runtime/issues/70762)
* [Enable compression on blazorpack](https://github.com/dotnet/aspnetcore/issues/35897)
* [[JSInterop] Little improvements](https://github.com/dotnet/aspnetcore/issues/31151)
* [Mechanism for delaying Blazor's first render until after some async operation](https://github.com/dotnet/aspnetcore/issues/45131)
* [Avoid completely replacing the DOM when components are re-rendered after prerendering](https://github.com/dotnet/aspnetcore/issues/42561)
* [Consider using C#9 Source Generators for implementing SetParametersAsync](https://github.com/dotnet/aspnetcore/issues/29550)
* [Ability to run multiple Blazor server / Web assembly apps in the same document](https://github.com/dotnet/aspnetcore/issues/38128)
* [Consider re-designing the Blazor WASM authentication stack](https://github.com/dotnet/aspnetcore/issues/40764)
* [Blazor Server: Provide APIs/extensibility for circuit eviction](https://github.com/dotnet/aspnetcore/issues/17866)
* [Runtime updates for WASM file size upload limit (the PU PR is directed at .NET 9)](https://github.com/dotnet/runtime/pull/91295)
* [Automate custom icu creation](https://github.com/dotnet/runtime/issues/82908)

## Other

Status | Live | Issue | PR | Item | PU Links/<br>Blog
:---: | :---: | :---: | :---: | --- | :---:
✔️ | &mdash; | #30869 #30871 | [`dotnet/blazor-samples/pull/112`](https://github.com/dotnet/blazor-samples/pull/112) #30872 | Non-BWA 8.0 `dotnet/blazor-samples` sample app updates.
✔️ | [What's New](https://learn.microsoft.com/en-us/aspnet/core/release-notes/aspnetcore-8.0?view=aspnetcore-8.0#blazor)<br>[Migration](https://learn.microsoft.com/en-us/aspnet/core/migration/70-80?view=aspnetcore-8.0&tabs=visual-studio)<br>[Integration](https://learn.microsoft.com/en-us/aspnet/core/blazor/components/integration?view=aspnetcore-8.0) | #29505 #30471 #29522 #30270 #30843 | #30671 #30791 #30799 #30279 #30817 #30799 #30845 #31019 | *What's New*, *Migration*, and *Integration* content| Will make a final pass near RTM on *What's New* guidance.
✔️ | [Live](https://learn.microsoft.com/en-us/aspnet/core/migration/70-80?view=aspnetcore-8.0#drop-blazor-server-with-yarp-routing-workaround) | #30528 | #30885 #30889 | Yarp with Blazor |
&mdash; | [Live](https://learn.microsoft.com/en-us/aspnet/core/blazor/security/webassembly/?view=aspnetcore-8.0) | 

## Blazor Hybrid

Status | Live | Issue | PR | Item | PU Links
:---: | :---: | :---: | :---: | --- | :---:
✔️ | [Live](https://learn.microsoft.com/en-us/aspnet/core/blazor/hybrid/troubleshoot?view=aspnetcore-8.0) | #30553 | #30554 | Diagnose issues in Blazor Hybrid | [dotnet/docs-maui #1207](https://github.com/dotnet/docs-maui/issues/1207)
✔️ | &mdash; | #28644 | #28673 | MAUI Blazor template name change | [PU Issue](https://github.com/dotnet/maui/issues/13804) VS4Mac portion dropped.
✔️ | [Live](https://learn.microsoft.com/en-us/aspnet/core/blazor/hybrid/index?view=aspnetcore-8.0#access-scoped-services-from-native-ui) | #30555 | #30556 | BlazorWebView gains a TryDispatchAsync method | [PU Issue](https://github.com/dotnet/docs-maui/issues/1422) [PU PR](https://github.com/dotnet/maui/pull/13425)
✔️ | [Live](https://learn.microsoft.com/en-us/aspnet/core/blazor/hybrid/routing?view=aspnetcore-8.0&pivots=maui#get-or-set-a-path-for-initial-navigation) | #28965 | #28968 | Specify initial URL for BlazorWebView to load | [PU issue](https://github.com/dotnet/maui/issues/11746) [PU PR](https://github.com/dotnet/maui/pull/11942)

## Moved to .NET 9

* Add .NET MAUI Blazor deep linking guidance - [PU issue](https://github.com/dotnet/maui/issues/3788#issuecomment-1421550198), [PU issue](https://github.com/dotnet/docs-maui/issues/1303), [Docs Issue](https://github.com/dotnet/AspNetCore.Docs/issues/28401), [Docs PR1 (remark)](), [Docs PR2 (remark update)](https://github.com/dotnet/AspNetCore.Docs/pull/30551)
* Backlogged: Intercepting requests from BlazorWebView maui - [PU Issue](https://github.com/dotnet/maui/issues/11382)

## Blog posts for cross-links

Marking "done" on these without addressing the *full-stack UI* updates because it will be for no earlier than Pre5.

* ***Done!*** 👍 https://devblogs.microsoft.com/dotnet/asp-net-core-updates-in-dotnet-8-preview-1/
* ***Done!*** 👍 https://devblogs.microsoft.com/dotnet/asp-net-core-updates-in-dotnet-8-preview-2/
* ***Done!*** 👍 https://devblogs.microsoft.com/dotnet/asp-net-core-updates-in-dotnet-8-preview-3/
* ***Done!*** 👍 https://devblogs.microsoft.com/dotnet/asp-net-core-updates-in-dotnet-8-preview-4/
* ***Done!*** 👍 https://devblogs.microsoft.com/dotnet/asp-net-core-updates-in-dotnet-8-preview-5/
* ***Done!*** 👍 https://devblogs.microsoft.com/dotnet/asp-net-core-updates-in-dotnet-8-preview-6/
* ***Done!*** 👍 https://devblogs.microsoft.com/dotnet/asp-net-core-updates-in-dotnet-8-preview-7/
* ***Done!*** 👍 https://devblogs.microsoft.com/dotnet/asp-net-core-updates-in-dotnet-8-rc-1/
* ***Done!*** 👍 https://devblogs.microsoft.com/dotnet/asp-net-core-updates-in-dotnet-8-rc-2/
* ***Done!*** 👍 https://devblogs.microsoft.com/dotnet/announcing-asp-net-core-in-dotnet-8/

## 8.0 preview/RC PU issues for analysis

This section of linked issues is assessed for items that didn't make the roadmap but have a bearing on docs. This section will be removed after the last batch for RC2 are assessed. RC1 and RC2 will be assessed after their milestone versions release.

* https://github.com/dotnet/aspnetcore/issues?q=is%3Aissue+label%3Aarea-blazor+milestone%3A8.0-preview1
* https://github.com/dotnet/aspnetcore/issues?q=is%3Aissue+label%3Aarea-blazor+milestone%3A8.0-preview2
* https://github.com/dotnet/aspnetcore/issues?q=is%3Aissue+label%3Aarea-blazor+milestone%3A8.0-preview3
* https://github.com/dotnet/aspnetcore/issues?q=is%3Aissue+label%3Aarea-blazor+milestone%3A8.0-preview4
* https://github.com/dotnet/aspnetcore/issues?q=is%3Aissue+label%3Aarea-blazor+milestone%3A8.0-preview5
* https://github.com/dotnet/aspnetcore/issues?q=is%3Aissue+label%3Aarea-blazor+milestone%3A8.0-preview6
* https://github.com/dotnet/aspnetcore/issues?q=is%3Aissue+label%3Aarea-blazor+milestone%3A8.0-preview7
* https://github.com/dotnet/aspnetcore/issues?q=is%3Aissue+label%3Aarea-blazor+milestone%3A8.0-rc1
* https://github.com/dotnet/aspnetcore/issues?q=is%3Aissue+label%3Aarea-blazor+milestone%3A8.0-rc2

## .NET 8 Planning

https://github.com/dotnet/aspnetcore/issues?q=is%3Aopen+is%3Aissue+label%3Aarea-blazor+milestone%3A%22.NET+8+Planning%22

Issues to watch :eyes: ...

* https://github.com/dotnet/aspnetcore/issues/50744
* https://github.com/dotnet/aspnetcore/issues/50645
* https://github.com/dotnet/aspnetcore/issues/45654
* https://github.com/dotnet/aspnetcore/issues/45627
* https://github.com/dotnet/aspnetcore/issues/45399
* https://github.com/dotnet/aspnetcore/issues/45394
* https://github.com/dotnet/aspnetcore/issues/45120
* https://github.com/dotnet/aspnetcore/issues/44866
* https://github.com/dotnet/aspnetcore/issues/28640
* https://github.com/dotnet/aspnetcore/issues/44143
* https://github.com/dotnet/aspnetcore/issues/43191
* https://github.com/dotnet/aspnetcore/issues/42216
* https://github.com/dotnet/aspnetcore/issues/42105
* https://github.com/dotnet/aspnetcore/issues/42003
* https://github.com/dotnet/aspnetcore/issues/40736
* https://github.com/dotnet/aspnetcore/issues/38842
* https://github.com/dotnet/aspnetcore/issues/38624
* https://github.com/dotnet/aspnetcore/issues/38552
* https://github.com/dotnet/aspnetcore/issues/38486
* https://github.com/dotnet/aspnetcore/issues/31922
* https://github.com/dotnet/aspnetcore/issues/31604
* https://github.com/dotnet/aspnetcore/issues/28938
* https://github.com/dotnet/aspnetcore/issues/28821
* https://github.com/dotnet/aspnetcore/issues/28770
* https://github.com/dotnet/aspnetcore/issues/28640
* https://github.com/dotnet/aspnetcore/issues/26676
* https://github.com/dotnet/aspnetcore/issues/25274
* https://github.com/dotnet/aspnetcore/issues/25152
* https://github.com/dotnet/aspnetcore/issues/18449
* https://github.com/dotnet/aspnetcore/issues/18088
* https://github.com/dotnet/aspnetcore/issues/18016
* https://github.com/dotnet/aspnetcore/issues/17986
* https://github.com/dotnet/aspnetcore/issues/12302
* https://github.com/dotnet/aspnetcore/issues/12158
* https://github.com/dotnet/aspnetcore/issues/10522
* https://github.com/dotnet/aspnetcore/issues/6921
* https://github.com/dotnet/aspnetcore/issues/6818
* https://github.com/dotnet/aspnetcore/issues/5523

## PU roadmap

[ASP.NET Core Roadmap for .NET 8 (dotnet/aspnetcore #44984)](https://github.com/dotnet/aspnetcore/issues/44984)
